### PR TITLE
[Azure] Fix the metrics field name in the container instance datastream

### LINKED ISSUE #7158: [Azure] [container_instance] Not correct metrics field
azure_metrics datastrems are using rename processor to rename `azure.metrics` to `azure.{datastream_name}`
https://github.com/elastic/integrations/blob/main/packages/azure_metrics/data_stream/container_registry/elasticsearch/ingest_pipeline/default.yml#L7-L10

For `container_instance` is used `azure.monitor` field - https://github.com/elastic/integrations/blob/main/packages/azure_metrics/data_stream/container_instance/elasticsearch/ingest_pipeline/default.yml#L7-L10, that does not exist.

As result metrics are reported under `azure.metrics`, instead of under `azure.container_instance`

### PR DESCRIPTION
**Pull request by zmoog**
## What does this PR do?

Fix the rename processor to use the correct field name. Add aliases to allow existing users to access all data with the old field names `azure.metrics.*` for backwards compatibility.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [x] Fix rename processor
- [x] Add aliases for the `azure.metrics.*` field names

### Aliases for backwards compatibility

Existing users can use field names `azure.metrics.*`in dashboards and queries to access old and new documents:

In the `metrics-azure.container_instance-default` there are:

- `12` documents created with integration v1.0.23, having `azure.metrics` fields only.
- `22` documents created with integration v1.0.24, having `azure.container_instance` fields only.

```
#
# request
#

POST metrics-azure.container_instance-default/_search
{
  "query": {
    "exists":{
      "field":"azure.metrics"
    }
  }
}

#
# response
#

{
  "hits": {
    "total": {
      "value": 34,
      "relation": "eq"
    },


#
# request
#

POST metrics-azure.container_instance-default/_search
{
  "query": {
    "exists":{
      "field":"azure.container_instance"
    }
  }
}

#
# response
#

{
  "hits": {
    "total": {
      "value": 22,
      "relation": "eq"
    },
```

If we query `azure.container_instance` we get the 22 new docs only. However, if we query using `azure.metrics` we get all 34 (12 + 22) documents.

## Related issues

- Closes #7158

## Screenshots

The dashboard can show data now that the pipeline creates the documents with the expected field names.

CleanShot 2023-08-17 at 16 44 36@2x

### CODE REVIEW

**Review by rajvi-elastic — APPROVED** [review]
Looks good to me!

**Review by tetianakravchenko — COMMENTED** [review]
changelog and manifest version update are missing, could you please add it?

maybe it makes sense to update the relevant dashboards with the correct metric naming? for example: If I install fixed version of integration, it might be confusing to find that used `azure.metrics...` instead of expected `azure.container_instance`, @zmoog wdyt?

**Comment by zmoog** [comment]
> changelog and manifest version update are missing, could you please add it?

🤦 

I added the missing changelog entry, thanks!


> maybe it makes sense to update the relevant dashboards with the correct metric naming? for example: If I install fixed version of integration, it might be confusing to find that used `azure.metrics...` instead of expected `azure.container_instance`, @zmoog wdyt?

The dashboards was already using `azure.container_instance.*` field names. Without this change, the dashboard does not display anything. I am surprised we didn't receive and SDH or other bug report about problem.

File: packages/azure_metrics/changelog.yml [hunk 1]
```diff
@@ -1,3 +1,8 @@
+- version: "1.0.24"
+  changes:
+    - description: Fix container instance metrics field problem
```
Comment (tetianakravchenko) [inline packages/azure_metrics/changelog.yml:3]: maybe rather `field problem` -> `field path` ?

**Review by tetianakravchenko — APPROVED** [review]

File: packages/azure_metrics/changelog.yml (hunk 1 above)
Comment (zmoog) [reply packages/azure_metrics/changelog.yml:3]: Good suggestion!

The description didn't sound completely right also for me, but I couldn't get a better one.

**Review by endorama — APPROVED** [review]
> Add aliases to allow existing users to access all data with the old field names azure.metrics.* for backwards compatibility.

🤩

**Comment by zmoog** [comment]
> > Add aliases to allow existing users to access all data with the old field names azure.metrics.* for backwards compatibility.
> 
> 🤩

If you break it, you fix it ™

**Pull request merged**